### ISSUE TITLE
"Subscribe for Pro" doesn't reactivate a cancelled subscription

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#9440]
I noticed this while poking around in stage; STR:

* Visit https://123done-stage.dev.lcip.org/ and "Subscribe for Pro". This all worked as expected.
* Visit https://accounts.stage.mozaws.net/settings, click subscriptions -> manage, and cancel the subscription that was just set up
* Visit https://123done-stage.dev.lcip.org/ and "Subscribe for Pro" again

The flow all seems to silently succeed, redirecting me back to 123done. But when I go into my account settings, the subscription is still cancelled. I expected it to ask me to re-activate the subscription.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-260)

**Comment by data-sync-user** [comment]
➤ Wil Clouser commented:

This should go through the normal subscribe process

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

tl;dr: Cancellation is not immediate - it's deferred to end of paid term. Since the subscription is still active (and already paid for), there's no subscription payment process we can offer the user until then.

> This should go through the normal subscribe process

Not sure what that means... The process where we ask for payment again? Or no change and just follow the current normal process where we see the user already has an active subscription and skip the payment form to redirect to the product?

We could try adding a check to see if the active subscription expires at end of term (i.e. the user cancelled it) and bounce to subscription management instead of the product page. They can reactivate from there. But, I don't think we can or should run the user through a payment flow again.

Could also consult UX to add a new flow specifically for attempting to subscribe when when already subscribed but pending cancellation at end of term.

**Comment by data-sync-user** [comment]
➤ Wil Clouser commented:

Benjamin Bangert - what happens today if a person cancels and then tries to resubscribe within the period they already paid for?

**Comment by data-sync-user** [comment]
➤ Benjamin Bangert commented:

The same thing that this issue documents occurs today. This is because the subscription is still active as Les mentioned above. His suggestion also still seems to be the best, bounce the user over to the management page so they could hit the resubscribe button.

**Comment by cknowles-admin** [comment]
FxA is no longer synchronizing all issues between Jira and Github.  We are closing open issues and will selectively synchronize in the future.
